### PR TITLE
Run the code formatter on Mix.Tasks.Compile.App

### PR DESCRIPTION
**Pull request by Nitrino**

### CODE REVIEW

File: lib/mix/lib/mix/tasks/compile.app.ex [hunk 1]
```diff
@@ -199,73 +208,122 @@ defmodule Mix.Tasks.Compile.App do
...
     included_applications = Keyword.get(properties, :included_applications, [])
 
-    for %{app: app, opts: opts, top_level: true} <- Mix.Dep.cached,
+    for %{app: app, opts: opts, top_level: true} <- Mix.Dep.cached(),
         Keyword.get(opts, :app, true),
         Keyword.get(opts, :runtime, true),
         not Keyword.get(opts, :optional, false),
-        app not in included_applications,
-        do: app
+        app not in included_applications, do: app
```
Comment (Nitrino) [inline lib/mix/lib/mix/tasks/compile.app.ex]: I'm not sure about location `do` block on one line with condition

File: lib/mix/lib/mix/tasks/compile.app.ex [hunk 2]
```diff
@@ -199,73 +208,122 @@ defmodule Mix.Tasks.Compile.App do
...
 
   defp validate_properties!(properties) do
-    Enum.each properties, fn
+    Enum.each(properties, fn
       {:description, value} ->
         unless is_list(value) do
-          Mix.raise "Application description (:description) is not a character list, got: #{inspect value}"
+          Mix.raise(
+            "Application description (:description) is not a character list, got: #{
+              inspect(value)
```
Comment (Nitrino) [inline lib/mix/lib/mix/tasks/compile.app.ex]: And I'm not sure about such splits of long lines with variables. How best to do in such cases?

Comment (josevalim) [reply lib/mix/lib/mix/tasks/compile.app.ex]:
```
msg = "Application description (:description) is not a character list, got: #{inspect(value)}"
```

or if it still doesn't fit, break using `<>`:

```elixir
"Application description (:description) is not a character list, " <>
  "got: #{inspect(value)}"
```

File: lib/mix/lib/mix/tasks/compile.app.ex (hunk 1 above)
Comment (josevalim) [reply lib/mix/lib/mix/tasks/compile.app.ex]: Good catch. This is a bug. I will investigate!

**Comment by josevalim** [comment]
Thanks @Nitrino! For the cases where the interpolation broke apart, it is best to use the concatenation operators. Regarding the `do: ...`, it is a bug, we will investigate and let you know once this is ready for reformatting.

**Comment by josevalim** [comment]
Hi @Nitrino! I have pushed a fix to master for your long comprehension. Please rebase on latest master, run `make compile` again and then format the file once more. Thanks! :heart:

**Comment by Nitrino** [comment]
@josevalim, after rebasing from master with your fix and recompile,  formatter still thinks that `do: ...` should be on the same line

**Comment by josevalim** [comment]
@Nitrino I have tried it locally and it worked. So I will merge it here and fix it locally but meanwhile please try this:

    git checkout master
    make clean compile

Thank you!  :heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by Nitrino** [comment]
@josevalim thanks, `make clean compile` helped

**Pull request merged**